src/RFC-0150_Wallets.md

```diff
@@ -61,58 +73,58 @@ This RFC is derived from a proposal first made in [this issue](https://github.co
-- Wallet features requiring querying a base node for information SHOULD have caching capabilities to reduce bandwidth consumption.
+- Wallet **MUST** be able to establish connection to the [base node][Base Node], submit transactions and monitor the Tari [blockchain].
+- Wallet **SHOULD** maintain an internal ledger to keep track of the Tari coin balance.
+- Wallet **MAY** offer transaction fee estimation, taking into account:
+  - transaction byte size
+  - network congestion
+  - desired transaction priority
+- Wallet **SHOULD** be able to monitor and present states (`Spent`, `Unspent` or `Unconfirmed`) of previously submitted transactions, 
+by querying information from the connected base node.
+- Wallet **SHOULD** present the total `Spent`, `Unspent` or `Unconfirmed` transactions in a summarized form. 
+- Wallet **SHOULD** be able to update its software to patch potential security vulnerabilities. 
+Automatic updating **SHOULD** be enabled by default, but users can decide to opt out.
+- Wallet **SHOULD** feature a caching mechanism for querying operations to reduce bandwidth consumption.
 
 #### Key Management Features
 
-- It MUST be able to generate a master seed key for the wallet by using:
-  - input from a user (e.g. when restoring a wallet, or in testing); or
-  - a user-defined set of mnemonic word sequences using known word lists; or
-  - a cryptographically secure random number generator.
-- It SHOULD be able to generate derived transactional cryptographic key pairs from the master seed key using deterministic 
-key pair generation.
-- It SHOULD store the wallet state using a password or passphrase encrypted persistent key-value database.
-- It SHOULD provide the ability to back up the wallet state to a single encrypted file to simplify wallet recovery and 
+- Wallet **MUST** be able to generate a master seed key for the wallet by using at least one of the following methods:
+  - input from the user (e.g. when restoring a wallet or in testing),
+  - user-defined set of mnemonic word sequences using known word lists,
+  - cryptographically secure random number generator.
+- Wallet **SHOULD** be able to generate derived, transactional, cryptographic keypairs from the master seed key using deterministic 
+keypair generation.
+- Wallet **SHOULD** store wallet state using a password or passphrase encrypted persistent key-value database.
+- Wallet **SHOULD** provide the ability to store backup of the wallet state to a single encrypted file to simplify wallet recovery and 
 reconstruction at a later stage.
-- It MAY provide the ability to export the master seed key or wallet state as a printable paper wallet, using coded markers.
+- Wallet **MAY** provide the ability to export the master seed key or the wallet state as a printable paper wallet, using coded markers.
 
-#### Different Methods for Recovering Wallet State of Tari Software Wallet
+#### Tari Software Wallet: State Recovery Methods
```

Comment:
I don't think we have a full detailed description of the wallet recovery (might be mistaken).
@stringhandler Do you think we should include a full description here? This RFC is probably intentionally very light on details so maybe we need a new RFC describing wallet recovery. We have bulletproof rewinding but not  the exact wallet recovery process. 

```suggestion
#### State Recovery
```